debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -238,6 +244,16 @@ public void getChangesForTables(SqlServerChangeTable[] changeTables, Lsn interva
         prepareQuery(queries, preparers, consumer);
     }
 
+    private String lsnTimestampSelectStatement() {
+        if (SourceTimestampMode.PROCESSING.equals(sourceTimestampMode)) {
+            return "";
+        }
+
+        return supportsAtTimeZone
+                ? ", sys.fn_cdc_map_lsn_to_time([__$start_lsn]) AT TIME ZONE 'UTC'"
+                : ", sys.fn_cdc_map_lsn_to_time([__$start_lsn])";
```

Comment:
These bits could be also extracted in static properties. Additionally, the duplication of `sys.fn_cdc_map_lsn_to_time([__$start_lsn])` could be eliminated.

Reply:
+1 to both.

Reply:
> processing mode interface

1. Why specificly _processing mode_? How about _commit mode_?
2. There is no such thing as `mode interface`. It's enum. In order move this logic to that layer we need to replace the enum with an interface, two classes and a factory method. Given that all the SQL manipulation logic is concentrated in this (`SqlServerConnection`) class is it reasonable to move a small bit of it out to a different place?

Reply:
>  Why specifically _processing mode_? How about _commit mode_?

Sorry, it was meant to be the "timestamp" mode (`SourceTimestampMode`). No need to introduce a new interface.

Reply:
I've extracted these sql parts to static properties and also have moved the logic to `SourceTimestampMode`.